### PR TITLE
[18SJ] Add special 2 player variant

### PR DESCRIPTION
**Pull request by perwestling**
This 2 player variant adds a 3rd player, a bot, that is
partly automatic, and partly handled by the players, via
master mode. The bot is excluded from the value/result.

More details in the 18SJ rule book.

### CODE REVIEW

File: lib/engine/game/g_18_sj/step/buy_train.rb [hunk 1]
```diff
@@ -19,6 +19,28 @@ def actions(entity)
           end
 
           def do_after_buy_train_action(_action, _entity); end
+
+          def buyable_trains(entity)
+            if entity.player == @game.edelsward
+              result = @depot.depot_trains[0..1]
+              puts result.to_s
```
Comment (tobymao) [inline lib/engine/game/g_18_sj/step/buy_train.rb:26]: remove debug

File: lib/engine/game/g_18_sj/step/buy_train.rb [hunk 2]
```diff
@@ -19,6 +19,28 @@ def actions(entity)
           end
 
           def do_after_buy_train_action(_action, _entity); end
+
+          def buyable_trains(entity)
+            if entity.player == @game.edelsward
+              result = @depot.depot_trains[0..1]
```
Comment (tobymao) [inline lib/engine/game/g_18_sj/step/buy_train.rb:25]: result is never used?

File: lib/engine/game/g_18_sj/step/buy_train.rb (hunk 1 above)
Comment (perwestling) [reply lib/engine/game/g_18_sj/step/buy_train.rb:26]: Fixed

File: lib/engine/game/g_18_sj/step/buy_train.rb (hunk 2 above)
Comment (perwestling) [reply lib/engine/game/g_18_sj/step/buy_train.rb]: This code was a remain from when I tried to get this to work.

The final way was that as part of the cleanup after dividend, a corporation owned (requisited 100%) by the bot (Edelswärd) automatically gets 0-2 trains without using buy_trains. So a bot will not use that step.

**Pull request merged**